iceoryx_utils/include/iceoryx_utils/internal/concurrent/lockfree_queue/index_queue.inl

```diff
@@ -209,6 +223,60 @@ bool IndexQueue<Capacity, ValueType>::popIfFull(ValueType& index) noexcept
     return false;
 }
 
+template <uint64_t Capacity, typename ValueType>
+bool IndexQueue<Capacity, ValueType>::popIfSizeIsAtLeast(uint64_t requiredSize, ValueType& index) noexcept
+{
+    if (requiredSize == 0)
+    {
+        return false;
+    }
+
+    // which to load first is up to discussion, for correctness it should make no difference
+    // but for performance it might
+    // note that without sync mechanisms (such as seq_cst), reordering is possible
+    auto writePosition = m_writePosition.load(std::memory_order_relaxed);
+    auto readPosition = m_readPosition.load(std::memory_order_relaxed);
+
+    // note: In principle it should be possible to load this just before the value
+    // is required in a successful return.
+    // However, doing this leads to failure of some stress tests for unkwown reasons
+    // This suggests that there is an issue which is not understood and needs to be investigated.
```

Comment:
Could you check what happens if you spin for a few CPU cycles right before this? Just in case it's a race which isn't triggered when you load it immediately after the read position ... assuming the compiler doesn't reorder

Reply:
I think I resolved this issue and can move the load, but I will look into it again. It was very weird and I remember having an open point in understanding of this underlying issue. We talked about it quite a while ago, but it was inconclusive.

Reply:
Resolved, I the issue I remember was caused by a defect in the test which I fixed a while ago (I think). Now loading later works (it also works with a sleep as you suggested). Here working means my stresstest passed multiple times (and never failed).

Hence I moved the load to a point where it is actually needed.